documentation/src/main/asciidoc/public/reference/_search-dsl-predicate.adoc

```diff
@@ -268,6 +268,42 @@ include::{sourcedir}/org/hibernate/search/documentation/search/predicate/Predica
 ----
 ====
 
+[[search-dsl-predicate-match-minimumshouldmatch]]
+=== `minimumShouldMatch`: fine-tuning how many terms are required to match
+
+It is possible to require that an arbitrary number of terms from the match string to be present in the document
+in order for the `match` predicate to match.
+This is the purpose of the `minimumShouldMatch*` methods,
+as demonstrated below.
+
+.Fine-tuning matching requirements with `minimumShouldMatch`
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=match-minimumShouldMatchNumber]
+----
+<1> At least two terms must match for this predicate to match.
+<2> All returned hits will match at least two of the terms:
+their titles will match either `investigation` and `detective`,
+`investigation` and `automatic`, `detective` and `automatic`, or all three of these terms.
+====
+
+[[search-dsl-predicate-match-boolean-operator]]
+=== Boolean operator
+
+By default, when using a `match` predicate, documents will match if *any* term mentioned in the match string is present in the document.
+Which is the behavior of an `OR` operator.
+Changing the operator to `AND` makes document match if *all* terms mentioned in the match string are present in the document
+Changing the operator can be done by calling `.operator(...)`.
+
+.`match` predicate with `AND` operator
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=match-operator]
+----
+====
+
```

Comment:
I'm not sure how this is related to this PR... but you definitely should have a look at https://hibernate.atlassian.net/browse/HSEARCH-917 . See the comments in particular.

You probably will want to have a look at https://hibernate.atlassian.net/browse/HSEARCH-3327 too, because I think Elasticsearch exposes a single settings controlling both multi-term handling and multi-field handling.

Reply:
> I'm not sure how this is related to this PR...

I've convinced myself before taking a look at the query builder that it uses a `MUST` clause and thought .. how would I test min should match if the clauses are must clauses? As in this comment 😄:

> From my experience with Hibernate Search, a lot of people are mistaken about how matching() works so it might be a good idea to give them an hint in the API.

But when I actually looked at the query builder, I already had the operator ready, so I thought I'd keep it and ask you what you think about it 😃

Reply:
I think it's a risky move if you don't have the whole problem in mind, but if you're confident, well... :)